api/cluster.go

```diff
@@ -27,6 +47,65 @@ import (
 //   400: Invalid data
 //   401: Unauthorized
 //   409: Cluster already exists
+func createCluster(w http.ResponseWriter, r *http.Request, t auth.Token) (err error) {
+	allowed := permission.Check(t, permission.PermClusterCreate)
+	if !allowed {
+		return permission.ErrUnauthorized
+	}
+	dec := form.NewDecoder(nil)
+	dec.IgnoreCase(true)
+	dec.IgnoreUnknownKeys(true)
+	var provCluster cluster.Cluster
+	err = r.ParseForm()
+	if err == nil {
+		err = dec.DecodeValues(&provCluster, r.Form)
+	}
+	if err != nil {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: err.Error(),
+		}
+	}
+	err = createClusterMachine(&provCluster)
```

Comment:
Shouldn't this be called only after you check that there isn't already a cluster with that name (line 84)?